### ISSUE TITLE
Secure boot "Verification Failed" error since v5.20230909.0 on Arch

### ISSUE THREAD

**Issue by sonnysighedup** [Limine-Bootloader/Limine#305]
Since release v5.20230909.0, I've been unable to use secure boot successfully with limine. I'm on an Arch-based variant, and up through v5.20230830.0 limine has worked flawlessly for me with secure boot enabled.

My updating process after each limine pkg upgrade:
1. Manually copy the updated BOOTX64.EFI to the appropriate place on my efi partition, calling it limine.efi
2. Signing limine.efi with sbsign
3. Enrolling the blake2b sum of my limine.cfg into limine.efi
4. Re-signing limine.efi with sbsign

What I have noticed in these latest upgrades and subsequent secure boot errors is that I no longer see a warning message like
`warning: data remaining[241592 vs 241664]: gaps between PE/COFF sections?`
after each instance of signing limine.efi

Any ideas on what might have changed from v5.20230830.0 that I need to adjust to?

Thank you for this great bootloader!

**Comment by Mintsuki** [comment]
Hi, thanks to you for the valuable feedback!

Is this by using `shim` or is this by directly signing Limine with a key you manually enrolled? Does this happen with `5.20230917.0` (I assume it still does)?

If this is by using `shim`, I suggest trying without it since Limine doesn't officially provide support for it and may be the source of issues like these.

**Comment by sonnysighedup** [comment]
Yes, I'm directly signing Limine with a manually enrolled key. I use `sbsign` from `sbsigntools`. My typical command is:
`doas sbsign --key /etc/efikeys/mok/MOK.key --cert /etc/efikeys/mok/MOK.crt --output /efi/EFI/BOOT/limine.efi /efi/EFI/BOOT/limine.efi`

I did try with `5.20230917.0` and still had the issue. Once I turn off secure boot, the booloader works. Downgrading to `5.20230830.0` and then re-enabling secure boot from BIOS makes everything goes back to normal.

**Comment by Mintsuki** [comment]
Do you have any idea whether the "Verification Failed" error comes from the Linux kernel, firmware/BIOS, or else?

**Comment by Mintsuki** [comment]
Because as far as I can tell "Verification Failed" seems to be a `shim` error, so please make sure you're not using shim.

**Comment by alicavus** [comment]
@sonnysighedup  why you are signing your bootloader twice?

**Comment by sonnysighedup** [comment]
> @sonnysighedup why you are signing your bootloader twice?

An unnecessary precaution. I don't know where I got the idea or developed the habit from. As it works either way, I will wean myself from the unnecessary extra step. Thanks!

**Comment by sonnysighedup** [comment]
> Because as far as I can tell "Verification Failed" seems to be a `shim` error, so please make sure you're not using shim.

You were right. I was using shim all along, and had forgotten that I resorted to it when I struggled to get secure boot working on my system back in March. I re-installed my secure boot keys et al, re-signed limine with the correct key and replaced shim.efi wtih limine in my UEFI stub. All is right with my world, and I'm running the latest version of limine with secure boot enabled.

Thank you for the good counsel, and sorry for not thinking things through properly before raising a fuss!
